File: app/src/main/java/ezike/tobenna/myweather/di/AppComponent.java
package ezike.tobenna.myweather.di;

import android.app.Application;

import javax.inject.Singleton;

import dagger.BindsInstance;
import dagger.Component;
import dagger.android.support.AndroidSupportInjectionModule;
import ezike.tobenna.myweather.WeatherApplication;
import ezike.tobenna.myweather.di.module.ActivityModule;
import ezike.tobenna.myweather.di.module.ApiModule;
import ezike.tobenna.myweather.di.module.AppModule;
import ezike.tobenna.myweather.di.module.DataSourceModule;
import ezike.tobenna.myweather.di.module.DatabaseModule;
import ezike.tobenna.myweather.di.module.FragmentBuildersModule;
import ezike.tobenna.myweather.di.module.LocationModule;
import ezike.tobenna.myweather.di.module.UnitModule;


/**
 * @author tobennaezike
 * @since 20/03/19
 */
@Singleton
@Component(modules = {
        AndroidSupportInjectionModule.class,
        ActivityModule.class,
        FragmentBuildersModule.class,
        ApiModule.class,
        DatabaseModule.class,
        AppModule.class,
        UnitModule.class,
        LocationModule.class,
        DataSourceModule.class,
        ViewModelModule.class})
public interface AppComponent {

    void inject(WeatherApplication application);

    @Component.Builder
    interface Builder {

        @BindsInstance
        Builder application(Application application);

        AppComponent build();
    }
}

File: app/src/main/java/ezike/tobenna/myweather/di/AppComponent.kt
package ezike.tobenna.myweather.di

import android.app.Application
import dagger.BindsInstance
import dagger.Component
import dagger.android.support.AndroidSupportInjectionModule
import ezike.tobenna.myweather.WeatherApplication
import ezike.tobenna.myweather.di.module.*
import javax.inject.Singleton

/**
 * @author tobennaezike
 * @since 20/03/19
 */
@Singleton
@Component(modules = [AndroidSupportInjectionModule::class,
    ActivityModule::class,
    ApiModule::class, DatabaseModule::class,
    AppModule::class, UnitModule::class,
    LocationModule::class,
    DataSourceModule::class,
    ViewModelModule::class])

interface AppComponent {
    fun inject(application: WeatherApplication?)

    @Component.Builder
    interface Builder {
        @BindsInstance
        fun application(application: Application): Builder

        fun build(): AppComponent
    }
}